### PR TITLE
Added Yandex provider

### PR DESCRIPTION
**Pull request by bologer**
| Q                        | A
| ------------------------ | ---
| Fixed Issues?            |  No
| Patch: Bug Fix?          | No
| Major: Breaking Change?  | No
| Minor: New Feature?      | Yes
| Documentation PR         | 

Added Yandex provider for v3 for #1013

### CODE REVIEW

File: src/Provider/Yandex.php [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function validateAccessTokenExchange($response)
+    {
+        $data = parent::validateAccessTokenExchange($response);
+
+        // Need to store user_id as token for later use.
```
Comment (ApacheEx) [inline src/Provider/Yandex.php]: do not see any usages of this. Why do we need to overried this method?

File: src/Provider/Yandex.php [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        if ($data->get('birthday')) {
+            list($birthday_year, $birthday_month, $birthday_day) = explode('-', $response->birthday);
+            $userProfile->birthDay = (int)$birthday_day;
+            $userProfile->birthMonth = (int)$birthday_month;
+            $userProfile->birthYear = (int)$birthday_year;
+        }
+
+        return $userProfile;
+    }
+}
```
Comment (ApacheEx) [inline src/Provider/Yandex.php]: add blank line

File: src/Provider/Yandex.php [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $accessTokenUrl = 'https://oauth.yandex.ru/token';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function hasAccessTokenExpired()
```
Comment (ApacheEx) [inline src/Provider/Yandex.php]: do not see reason to override this method - the same logic as in parent class.

**Comment by bologer** [comment]
@ApacheEx added changes, tested Providers, still works as expected :+1:

**Comment by bologer** [comment]
@ApacheEx could you please review and merge?

**Comment by ApacheEx** [comment]
Thanks a lot. Great job 🥇

**Pull request merged**